go.mod

```diff
@@ -1,120 +1,128 @@
 module github.com/nutanix-cloud-native/cloud-provider-nutanix
 
-go 1.22.0
+go 1.23
```

Comment:
Does updating k8s.io to v1.31 require using go 1.23? I see the openshift/machine-api-operator repo still uses go 1.22 and updated k8s.io to v1.31.

Reply:
Are we using something from this operator as a library in Openshift?

Reply:
Synced with OCP folks, and we can keep the go.mod version change in a separate PR to make it easy for them to filter out that PR.